### PR TITLE
refactor generateExtNameAssignment to reduce complexity

### PR DESCRIPTION
**Pull request by wanghbxxxx**
## What is the purpose of the change

XXXXX

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/AdaptiveClassCodeGenerator.java [hunk 1]
```diff
@@ -241,41 +241,14 @@ private String generateExtNameNullCheck(String[] value) {
...
-                        if (hasInvocation) {
-                            getNameCode = String.format("url.getMethodParameter(methodName, \"%s\", \"%s\")", value[i], defaultExtName);
-                        } else {
-                            getNameCode = String.format("url.getParameter(\"%s\")", value[i]);
-                        }
-                    } else {
-                        getNameCode = "url.getProtocol()";
-                    }
-                }
+            if ("protocol".equals(value[i])) {
```
Comment (khanimteyaz) [inline dubbo-common/src/main/java/org/apache/dubbo/common/extension/AdaptiveClassCodeGenerator.java]: use dubbo `Constants.PROTOCOL_KEY` instead  **protocol**

**Comment by khanimteyaz** [comment]
I believe UT are verified, if not or UT not present could you plz add those. Thanks for simplifying the implementation 👍

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/AdaptiveClassCodeGenerator.java (hunk 1 above)
Comment (wanghbxxxx) [reply dubbo-common/src/main/java/org/apache/dubbo/common/extension/AdaptiveClassCodeGenerator.java]: thanks

**Comment by wanghbxxxx** [comment]
> I believe UT are verified, if not or UT not present could you plz add those. Thanks for simplifying the implementation 👍

actually there is no UT at all, it is covered by UT of ExtensionLoader.
i'd like to add some UT which is not easy, because there is only one public method which does a lot.
maybe i could make some of private methods public/protected.
could you give me some advice please!
thank you

**Comment by AlbumenJ** [comment]
@wanghbxxxx hi, thanks for your contribution

pls resolve conflicts with the latest master branch

**Comment by AlbumenJ** [comment]
Close for long time no response.
Please feel free to reopen if you have any question.

**Pull request closed without merging**